teamapps-ux/src/main/java/org/teamapps/ux/session/SessionContext.java

```diff
@@ -278,7 +316,9 @@ public Template getTemplate(String id) {
 	}
 
 	/**
-	 * @param runnable
+	 * Executes the runnable within a thread that is bound to the SessionContext.
+	 * This ensures thread safety by letting only one thread interact with a UI-session at a time.
```

Comment:
Please also mention that the Runnable/Callable is not necessarily executed asynchronously but might actually be executed synchronously if the thread happens to be already bound to this SessionContext. 

I also think that it makes sense to add the JavaDoc comment to `runWithContext(Callable<R> runnable, boolean forceEnqueue)`.